### PR TITLE
fix parametrized routing

### PR DESCRIPTION
**Pull request by SuddenGunter**
- Fix issues with parametrized routes not being matched correctly in some edge cases.
- Added tests

### CODE REVIEW

File: radix/node.go [hunk 1]
```diff
@@ -129,14 +129,21 @@ func (n Node) Insert(path string, key uint64) Node {
 			return n
 		}
 
-		n.children = append([]Node{
-			{
+		if len(n.children) > 0 {
+			n.children = append(n.children, Node{
```
Comment (makasim) [inline radix/node.go:133]: hm, for it all if-else construct equivalent to:

```
n.children = append(n.children, Node{
	path: path[:start],
	children: []Node{
		updateParamNode(Node{kind: param}, path[start:], key),
	},
}
```

**Comment by makasim** [comment]
Have you checked all places `children` silce is modified ?

**Comment by SuddenGunter** [comment]
> Have you checked all places `children` silce is modified ?

imo they might not be relevant to this issue - I wasn't able to reproduce the issue anymore with this fix. If there are non-parametrized route being added, for example, order does not really matter there as much

File: radix/node.go (hunk 1 above)
Comment (makasim) [reply radix/node.go:133]: @SuddenGunter dont forget about this one ^^

Comment (SuddenGunter) [reply radix/node.go]: fixed

**Comment by makasim** [comment]
Thank you for contribution
